src/libraries/Common/tests/System/Diagnostics/Tracing/TestEventListener.cs

```diff
@@ -109,8 +111,14 @@ public async Task RunWithCallbackAsync(Action<EventWrittenEventArgs> handler, Fu
             finally { _eventWritten = null; }
         }
 
+        // Workaround for being able to inspect the ActivityId property after storing EventWrittenEventArgs
+        // [ActiveIssue("https://github.com/dotnet/runtime/issues/42128")]
+        private static readonly FieldInfo s_activityIdFieldInfo =
+            typeof(EventWrittenEventArgs).GetField("m_activityId", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
What is someone else supposed to do here?  We definitely don't want to encourage using private reflection into runtime internals, and we should be testing the way we'd encourage others to.  Is this something that no one else is going to need to do?

Reply:
This is a test-only workaround for the bug #42128 that should be fixed for 6.0.
The alternative to this workaround would be to change every test requiring the ActivityId to save the Guid from the callback itself - storing a `Queue<(EventArgs, Guid)>` instead. I found this change to be smaller than complicating all tests.

Reply:
> What is someone else supposed to do here?

For 5.0:
- do all the processing requiring the ActivityId of a non-Start/Stop event inside the callback itself or
- save the Guid outside of the EventArgs object itself inside the callback

Reply:
If we expect other developers to need to do the same operation while listening to our telemetry, I'd like for us to test it in a way we'd recommend they consume it.  If you don't believe anyone else will need to consume the telemetry in this manner, then ok.  If you do believe they will, then I'd like us to use one of the other approaches that avoids private reflection into corelib.

Reply:
If a user stored `EventWrittenEventArgs` instances for later post-processing (what we do tests), then they would hit this bug. In theory the approach is valid, short of a bug in 5.0. The recommended workaround for 5.0 would be to either a) not store the objects themselves, or b) fetch and save the ActivityId Guid while inside the callback. Both approaches boil down to storing something else than just `EventWrittenEventArgs`.
By 'storing' here I just mean processing the events on a different thread than the one executing the callback - I could see a `Channel<EventWrittenEventArgs>` being used for example.

For our test code, the only workaround would be storing the Guid manually at every call site.

By using the reflection workaround here, I was aiming at not having to change all tests even after the bug is fixed - the only change would be to remove the 3 lines of reflection in the TestEventListener.

For the sockets tests, the diff would be something like [this](https://gist.github.com/MihaZupan/5eae3dec1520be350782bc024998e3ae/revisions#diff-dcac16b3100ee0388ebcd74408915a3d), which feels to me like a lot of noise just to have it reverted later.

Reply:
It may be noise, but if this is a valid use case, when someone asks how to do it we should be able to point them to our tests as a good example and say "here's the recommended workaround".  I don't want private reflection into corelib to be the recommended workaround.

It may be a test for us, but it could very well be production code for someone else.  I'd prefer we do it the "right way".

Reply:
Changed (a39d49b)